lib/matplotlib/backends/registry.py

```diff
@@ -61,10 +198,75 @@ def backend_for_gui_framework(self, framework):
 
         Returns
         -------
-        str
-            Backend name.
+        str or None
+            Backend name or None if GUI framework not recognised.
         """
-        return self._GUI_FRAMEWORK_TO_BACKEND_MAPPING.get(framework)
+        return self._GUI_FRAMEWORK_TO_BACKEND.get(framework.lower())
+
+    def is_valid_backend(self, backend: str) -> bool:
```

Comment:
No need for inline typing if it's in the `.pyi`.

Reply:
Done